src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -572,16 +570,27 @@ protected Object clone() throws CloneNotSupportedException {
 
      *
      * @apiNote
-     * Reference processing or finalization may occur whenever the virtual machine detects that no
-     * reference to an object will ever be stored in the heap: The garbage
-     * collector may reclaim an object even if the fields of that object are
-     * still in use, so long as the object has otherwise become unreachable.
-     * This may have surprising and undesirable effects in cases such as the
-     * following example in which the bookkeeping associated with a class is
+     * Reference processing or finalization may occur when the virtual machine
+     * detects that there is no further need for an object. The garbage collector
+     * may reclaim an object even if values from that object's fields are still
+     * in use, or while a method on the object is still running, so long as the
+     * object has otherwise become unreachable.
+     * <p>
+     * This may have surprising and undesirable effects, in particular when using
```

Comment:
Nit: I don't think a new paragraph is appropriate here as "This may ..." refers back to the subject of the preceding lines.